lib/src/git_backend.rs

```diff
@@ -578,6 +579,19 @@ fn commit_from_git_without_root_parent(
             .map(|oid| CommitId::from_bytes(oid.as_bytes()))
             .collect_vec()
     };
+    // If the commit is a conflict, the conflict labels are stored in a commit
+    // header separately from the trees.
+    let conflict_labels: ConflictLabels = commit
+        .extra_headers()
+        .find(JJ_CONFLICT_LABELS_COMMIT_HEADER)
+        .map(|header| {
+            str::from_utf8(header)
+                .expect("labels should be valid utf8")
+                .split_terminator('\n')
```

Comment:
Can Git commit headers contain newlines? Could you update the test case to include newlines?